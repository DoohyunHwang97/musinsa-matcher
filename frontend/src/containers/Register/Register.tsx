import React, { useState } from "react";
import axios from "axios";
import "./Register.css"
axios.defaults.xsrfCookieName = "csrftoken";
axios.defaults.xsrfHeaderName = "X-CSRFToken";

const Register = () => {
  const [profile, setProfile] = useState<any>({
    username: "",
    password: "",
    nickname: "",
    email: "",
    length: "",
    waist_size: "",
    thigh_size: "",
    calf_size: "",
  });
  const handleInform = (e: any) => {
    setProfile((prevState: any) => {
      return { ...prevState, [e.target.name]: e.target.value };
    });
  };
  const clickSubmit = (e: any) => {
    e.preventDefault();
    console.log(profile);
    const user = {
      username: profile.username,
      password: profile.password,
      nickname: profile.nickname,
      email: profile.email,
      length: profile.length,
      waist_size: profile.waist_size,
      thigh_size: profile.thigh_size,
      calf_size: profile.calf_size,
    };
    axios
      .post("http://127.0.0.1:8000/api/clothes/signup/", user)
      .then((res: any) => {
        if (res) {
          localStorage.clear();
          localStorage.setItem("token", res.data.session_key);
          window.location.replace("/");
        } else {
          setProfile({
            username: "",
            password: "",
            nickname: "",
            email: "",
            length: "",
            waist_size: "",
            thigh_size: "",
            calf_size: "",
          });
        }
      })
      .catch((err: any) => {
        console.log(err.message);
        alert("wrong");
      });
  };
  return (
    <div>
      <div>
        <h2>Register</h2>
        <p>
          ID:
          <input
          className="RegisterInput"
            data-testid="username"
            id="username"
            name="username"
            onChange={handleInform}
          />
        </p>
        <p>
          Password:
          <input
          className="RegisterInput"
            id="password"
            type="password"
            name="password"
            onChange={handleInform}
          />
        </p>
        <p>
          Nickname:
          <input className="RegisterInput" id="nickname" name="nickname" onChange={handleInform} />
        </p>
        <p>
          Email:
          <input className="RegisterInput" id="email" name="email" onChange={handleInform} />
        </p>
      </div>
      <div>
        <h2>size</h2>
        <p>
          length:
          <input className="RegisterInput" id="length" name="length" onChange={handleInform} />
        </p>
        <p>
          waist size:
          <input className="RegisterInput" id="waist" name="waist_size" onChange={handleInform} />
        </p>
        <p>
          thigh size:
          <input className="RegisterInput" id="thigh" name="thigh_size" onChange={handleInform} />
        </p>
        <p>
          calf size:
          <input className="RegisterInput" id="calf" name="calf_size" onChange={handleInform} />
        </p>
      </div>
      <button className="button" data-testid="submit" onClick={clickSubmit}>submit</button>
    </div>
  );
};
export default Register;
