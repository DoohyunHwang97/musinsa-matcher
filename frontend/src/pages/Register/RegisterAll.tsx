import React, { useState } from "react";
import Axios from "axios";

const RegisterAll = () => {
  const [profile, setProfile] = useState<any>({
    username: "",
    password: "",
    nickname: "",
    email: "",
    chest_size: "",
    waist_size: "",
    thigh_size: "",
    calf_size: "",
  });
  const handleInform = (e: any) => {
    setProfile((prevState: any) => {
      return { ...prevState, [e.target.name]: e.target.value };
    });
  };
  const clickSubmit = (e: any) => {
    e.preventDefault();
    console.log(profile);
    const user = {
      username: profile.username,
      password: profile.password,
      nickname: profile.nickname,
      email: profile.email,
      chest_size: profile.chest_size,
      waist_size: profile.waist_size,
      thigh_size: profile.thigh_size,
      calf_size: profile.calf_size,
    };
    Axios.post("../../../public/test.json", user)
      .then((res: any) => {
        if (res.data.key) {
          localStorage.clear();
          localStorage.setItem("token", res.data.key);
          window.location.replace("/");
        } else {
          setProfile({
            username: "",
            password: "",
            nickname: "",
            email: "",
            chest_size: "",
            waist_size: "",
            thigh_size: "",
            calf_size: "",
          });
        }
      })
      .catch((err: any) => {
        console.log(err.message);

        alert("wrong");
      });
  };
  return (
    <div>
      <div>
        <h2>profile</h2>
        <p>
          ID:
          <input name="username" onChange={handleInform} />
        </p>
        <p>
          Password:
          <input type="password" name="password" onChange={handleInform} />
        </p>
        <p>
          Nickname:
          <input name="nickname" onChange={handleInform} />
        </p>
        <p>
          Email:
          <input name="email" onChange={handleInform} />
        </p>
      </div>
      <div>
        <h2>size</h2>
        <p>
          chest size:
          <input name="chest_size" onChange={handleInform} />
        </p>
        <p>
          waist size:
          <input name="waist_size" onChange={handleInform} />
        </p>
        <p>
          thigh size:
          <input name="thigh_size" onChange={handleInform} />
        </p>
        <p>
          calf size:
          <input name="calf_size" onChange={handleInform} />
        </p>
      </div>
      <button onClick={clickSubmit}>submit</button>
    </div>
  );
};
export default RegisterAll;
