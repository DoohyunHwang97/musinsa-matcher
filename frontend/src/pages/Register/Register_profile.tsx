import React, { useState, useEffect, useRef } from "react";
const Registerprofile = () => {
  const [profile, setProfile] = useState<any>({
    username: "",
    password: "",
    nickname: "",
    email: "",
  });
  const profileRef = useRef(profile);
  useEffect(() => {
    profileRef.current = profile;
  }, [profile]);
  useEffect(() => {
    return () => {
      console.log("unmount 시 출력", profileRef.current);
    };
  }, []);
  const handleInform = (e: any) => {
    setProfile((prevState: any) => {
      return { ...prevState, [e.target.name]: e.target.value };
    });
  };

  return (
    <div>
      <div>
        <h2>profile</h2>
        <p>
          ID:
          <input name="username" onChange={handleInform} />
        </p>
        <p>
          Password:
          <input name="password" onChange={handleInform} />
        </p>
        <p>
          Nickname:
          <input name="nickname" onChange={handleInform} />
        </p>
        <p>
          Email:
          <input name="email" onChange={handleInform} />
        </p>
      </div>
    </div>
  );
};
export default Registerprofile;
