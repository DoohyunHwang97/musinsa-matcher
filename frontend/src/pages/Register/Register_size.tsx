import React from "react";
const Registersize = () => {
  return (
    <div>
      <h2>size</h2>
      <p>
        size:
        <input />
      </p>
      <p>
        size:
        <input />
      </p>
      <p>
        size:
        <input />
      </p>
      <p>
        size:
        <input />
      </p>
    </div>
  );
};
export default Registersize;
