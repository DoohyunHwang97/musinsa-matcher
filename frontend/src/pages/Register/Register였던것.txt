import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import Registerprofile from "./Register_profile";
import Registersize from "./Register_size";

const Register = () => {
  const navigate = useNavigate();
  const regList: any = {
    0: <Registerprofile />,
    1: <Registersize />,
  };
  const [list, setList] = useState<number>(0);
  const clickHandler = (id: number) => {
    setList(id);
  };
  return (
    <>
      <div>
        <button onClick={() => navigate("/")}>Home</button>
      </div>

      <h2>Register</h2>
      <div>{regList[list]}</div>
      <button onClick={() => clickHandler(0)}>back</button>
      <button onClick={() => clickHandler(1)}>next</button>
      <div>
        <button>submit</button>
      </div>
    </>
  );
};
export default Register;
