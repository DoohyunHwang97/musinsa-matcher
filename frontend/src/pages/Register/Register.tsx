import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import Registerprofile from "./Register_profile";
import Registersize from "./Register_size";
import RegisterAll from "./RegisterAll";

const Register = () => {
  const navigate = useNavigate();
  const regList: any = {
    0: <Registerprofile />,
    1: <Registersize />,
  };
  const [list, setList] = useState<number>(0);
  const clickHandler = (id: number) => {
    setList(id);
  };
  return (
    <>
      <div>
        <button onClick={() => navigate("/")}>Home</button>
      </div>

      <h2>Register</h2>
      <div>
        <RegisterAll />
      </div>
    </>
  );
};
export default Register;
