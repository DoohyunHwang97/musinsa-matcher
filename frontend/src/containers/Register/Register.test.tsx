import React from "react";
import renderer from "react-test-renderer";
import { expect } from "@jest/globals";
import { shallow } from "enzyme";
import Register from "./Register";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Register", () => {
  let component: any = null;

  it("initial render", () => {
    component = renderer.create(<Register />);
  });
  it("render correct", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("성공적으로 렌더링되어야 합니다.", () => {
    const wrapper = shallow(<Register />);
    expect(wrapper.length).toBe(1);
  });
  it("should be change", () => {
    const wrapper: any = shallow(<Register />);
    wrapper.find("#username").simulate("change", { target: { value: "값" } });
    expect(wrapper.state().profile.username).toBe("값");
  });
});
