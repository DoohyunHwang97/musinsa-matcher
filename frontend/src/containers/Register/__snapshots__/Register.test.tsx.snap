// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register render correct 1`] = `
<div>
  <div>
    <h2>
      profile
    </h2>
    <p>
      ID:
      <input
        name="username"
        onChange={[Function]}
      />
    </p>
    <p>
      Password:
      <input
        name="password"
        onChange={[Function]}
        type="password"
      />
    </p>
    <p>
      Nickname:
      <input
        name="nickname"
        onChange={[Function]}
      />
    </p>
    <p>
      Email:
      <input
        name="email"
        onChange={[Function]}
      />
    </p>
  </div>
  <div>
    <h2>
      size
    </h2>
    <p>
      chest size:
      <input
        name="chest_size"
        onChange={[Function]}
      />
    </p>
    <p>
      waist size:
      <input
        name="waist_size"
        onChange={[Function]}
      />
    </p>
    <p>
      thigh size:
      <input
        name="thigh_size"
        onChange={[Function]}
      />
    </p>
    <p>
      calf size:
      <input
        name="calf_size"
        onChange={[Function]}
      />
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    submit
  </button>
</div>
`;
